src/components/verification-grid/verification-grid.ts

```diff
@@ -217,8 +217,20 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
 
   protected async updated(change: PropertyValueMap<this>): Promise<void> {
     const renderInvalidationKeys: (keyof this)[] = ["gridSize"];
-    const sourceInvalidationKeys: (keyof this)[] = ["getPage"];
     const tileInvalidationKeys: (keyof this)[] = ["selectionBehavior"];
+    // gridSize is a part of source invalidation because if the grid size
+    // increases, there will be verification grid tiles without any source
+    // additionally, if the grid size is decreased, we want the "currentPage"
+    // of sources to update / remove un-needed items
+    const sourceInvalidationKeys: (keyof this)[] = ["getPage", "gridSize"];
```

Comment:
I'm not sure on the scope of this... but this smells. I haven't read far enough into this diff to fully understand

I wonder if we're trying too hard here?  If extra items are added to the grid we render the page again. Any items that have the same source shouldn't change, and any items that are new get assigned a new source.

Reply:
Our source invalidation is optimized with a diff algorithm that will not re-render a tile/spectrogram if it has the same source.

By causing a source invalidation, only the new grid tiles that have been added will be affected

Reply:
seems like the wrong this was optimized.  A verification tile should do nothing if the same subjectwrapper is assigned to it. We shouldn't need the optimization at all. Still, out of scope. Leave a todo?

Reply:
In a previous PR (#104), we guaranteed that the memory address of the `SubjectWrapper` would not change throughout the program's lifetime.

So we now get this optimization through the Lit property decorators (for free).
This means we don't need the aforementioned diff optimization anymore
https://github.com/ecoacoustics/web-components/blob/main/src/components/verification-grid/verification-grid.ts#L1025-L1032